### PR TITLE
Update TeamsFx-preview-and-customize-app-manifest.md

### PR DESCRIPTION
**Pull request by yuqizhou77**
- Edit "customize app manifest" section. Teams Toolkit will support `file` function in manifest starting 5.10.

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md](https://github.com/MicrosoftDocs/msteams-docs/blob/cbd9091cc44d71a99c42c23f937516273215c4da/msteams-platform%2Ftoolkit%2FTeamsFx-preview-and-customize-app-manifest.md) | 94 | 11 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/6d947254-4353-414c-85dd-03fd5ced3f01) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md](https://github.com/MicrosoftDocs/msteams-docs/blob/fd56f97f3ab78dc9a3155122c6ada93b401425c2/msteams-platform%2Ftoolkit%2FTeamsFx-preview-and-customize-app-manifest.md) | 94 | 11 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/de36c83d-0e96-4a52-8240-2b61ccd9b8fa) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md](https://github.com/MicrosoftDocs/msteams-docs/blob/a61f0115bd13047aa1c5e8cee89c4d68814686b0/msteams-platform%2Ftoolkit%2FTeamsFx-preview-and-customize-app-manifest.md) | 94 | 10 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/89da7bc6-34cc-48b7-aaea-a194ec005af7) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md](https://github.com/MicrosoftDocs/msteams-docs/blob/bdc1843f62ee1a83ffdb469e426ab05f69a5272b/msteams-platform%2Ftoolkit%2FTeamsFx-preview-and-customize-app-manifest.md) | 94 | 10 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/b1499156-aff5-4a41-b478-674664082b9d) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 1]
```diff
@@ -76,7 +75,52 @@ For example, you can customize app description by defining a new environment var
             "full": "Full description of tab0418"
         },
     }
-```
+   ```
+
+1. Since Teams Toolkit 5.10, if you would like to store the value of a field, such as a lengthy or multiline app description, in a separate text file, you could use the `file` function. For example, you can create description.txt file sotring the full description of your app in the parent folder of `manifest.json`. Then set the value of description.full as `$[file('description.txt')]` in `manifest.json`. Teams Toolkit will read the content from the text file and use it as full description when building app package zip.
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:80]:
```suggestion
1. Starting with Teams Toolkit 5.10, using the `file` function you can store the value of a field, such as a lengthy or multiline app description, in a separate text file. Teams Toolkit will read the content from the text file and use it as full description when building an app package.
   For example, create a `description.txt` file in the parent folder of `manifest.json` to store your app's complete description. Then, set the value of `description.full` in `manifest.json` as `$[file('description.txt')]`.

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 2]
```diff
@@ -76,7 +75,52 @@ For example, you can customize app description by defining a new environment var
...
+        "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.16/MicrosoftTeams.schema.json",
+        "manifestVersion": "1.16",
+        "description": {
+            "short": "Short description of tab",
+            "full": "$[file('./description.txt')]"
+        },
+    }
+   ```
+
+   You could either put the file path in env/.env.xx and update the input parameter function of file function to placeholder in ${{xx_xx}} format.
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:101]:
```suggestion
To add the file path, update it in either `env` or `.env.xx`. Then, modify the `file` function to serve as a placeholder in the `${{xx_xx}}` format for the input parameter function.
```

**Review by v-npaladugu — CHANGES_REQUESTED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md](https://github.com/MicrosoftDocs/msteams-docs/blob/f5a101b9bb24a02332337015bc00fddd46c1eeab/msteams-platform%2Ftoolkit%2FTeamsFx-preview-and-customize-app-manifest.md) | 94 | 10 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/d02f5147-0682-41a5-af46-33aecebe9644) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md (hunk 2 above)
Comment (yuqizhou77) [reply msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:102]: This is another way user can set the parameter of file function. One way is to use it as "$[file('./description.txt')]" like what we introduced above. Another one, which we are introducing here, is to use environment variable. env/.env.xx is one file, not env or .env.xx. Could you help modify the string again?

Maybe something like "Or you could put the file path in env/.env.xx. Then, modify the parameter of `file` function to a placeholder in the `${{xx_xx}}` format."?

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md (hunk 1 above)
Comment (yuqizhou77) [reply msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md]: I think we should put "Teams Toolkit will read the content from the text file and use it as full description when building an app package." after the example as this is specific to explaining the full.description.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md](https://github.com/MicrosoftDocs/msteams-docs/blob/a34f3302b03c4b456f704401e4cd3d9e10408cf1/msteams-platform%2Ftoolkit%2FTeamsFx-preview-and-customize-app-manifest.md) | 94 | 10 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/61c7245b-d02d-4dc4-91cf-29ecdaf00f2f) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 3]
```diff
@@ -76,7 +75,52 @@ For example, you can customize app description by defining a new environment var
...
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:101]:
```suggestion
You can also add the file path in `env/.env.xx`. Then, modify the parameter of file function to a placeholder in the `${{xx_xx}}` format.
```

Comment (v-npaladugu) [reply msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:101]: @yuqizhou77

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 4]
```diff
@@ -76,7 +75,52 @@ For example, you can customize app description by defining a new environment var
...
+    ```text
+     DESCRIPTION_PATH=./description.txt
+    ```
+
+    `manifest.json`
+
+    ```json
+    {
+        "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.16/MicrosoftTeams.schema.json",
+        "manifestVersion": "1.16",
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:115]:
```suggestion
        "manifestVersion": "1.17",
```

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 5]
```diff
@@ -76,7 +75,52 @@ For example, you can customize app description by defining a new environment var
...
+   ```text
+    This is the full description.
+   ```
+
+   `manifest.json`
+
+   ```json
+    {
+        "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.16/MicrosoftTeams.schema.json",
+        "manifestVersion": "1.16",
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:93]:
```suggestion
        "manifestVersion": "1.17",
```

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 6]
```diff
@@ -76,7 +75,52 @@ For example, you can customize app description by defining a new environment var
...
+
+   ```text
+    This is the full description.
+   ```
+
+   `manifest.json`
+
+   ```json
+    {
+        "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.16/MicrosoftTeams.schema.json",
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:92]:
```suggestion
        "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.17/MicrosoftTeams.schema.json",
```

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 7]
```diff
@@ -59,15 +59,14 @@ For example, you can put your manifest.json file in `test/test.json`, and update
...
+   ```
+   `manifest.json`
 
-`manifest.json`
-
-```json
+   ```json
     {
         "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.16/MicrosoftTeams.schema.json",
         "manifestVersion": "1.16",
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:72]:
```suggestion
        "manifestVersion": "1.17",
```

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 8]
```diff
@@ -59,15 +59,14 @@ For example, you can put your manifest.json file in `test/test.json`, and update
...
+   TEAMS_APP_DESCRIPTION=This is an amazing app
+   ```
+   `manifest.json`
 
-`manifest.json`
-
-```json
+   ```json
     {
         "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.16/MicrosoftTeams.schema.json",
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:71]:
```suggestion
        "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.17/MicrosoftTeams.schema.json",
```

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 9]
```diff
@@ -76,7 +75,52 @@ For example, you can customize app description by defining a new environment var
...
+
+    ```text
+     DESCRIPTION_PATH=./description.txt
+    ```
+
+    `manifest.json`
+
+    ```json
+    {
+        "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.16/MicrosoftTeams.schema.json",
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:114]:
```suggestion
        "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.17/MicrosoftTeams.schema.json",
```

**Review by v-npaladugu — CHANGES_REQUESTED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md](https://github.com/MicrosoftDocs/msteams-docs/blob/8124859e45e5094e866b0e73278ff2a8a7831b2a/msteams-platform%2Ftoolkit%2FTeamsFx-preview-and-customize-app-manifest.md) | 94 | 10 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/52c80b9e-3a53-46b7-9735-699a07aac65a) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md](https://github.com/MicrosoftDocs/msteams-docs/blob/e89f90265c4744a3761443108d60d306bb144eb8/msteams-platform%2Ftoolkit%2FTeamsFx-preview-and-customize-app-manifest.md) | 94 | 10 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/24a431ba-3ef9-4da4-b6b9-2f11bbda5e7f) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 10]
```diff
@@ -59,24 +59,68 @@ For example, you can put your manifest.json file in `test/test.json`, and update
...
+        "manifestVersion": "1.17",
         "description": {
             "short": "${{TEAMS_APP_DESCRIPTION}}",
             "full": "Full description of tab0418"
         },
     }
-```
+   ```
+
+1. Starting with Teams Toolkit 5.10, using the `file` function you can store the value of a field, such as a lengthy or multiline app description, in a separate text file. For example, create a `description.txt` file in the parent folder of `manifest.json` to store your app's complete description. Then, set the value of `description.full` in `manifest.json` as `$[file('description.txt')]`. Teams Toolkit will read the content from the text file and use it as full description when building an app package.
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:80]:
```suggestion
1. Starting with Teams Toolkit 5.10, using the `file` function you can store the value of a field, such as a lengthy or multiline app description, in a separate text file. For example, create a `description.txt` file in the parent folder of `manifest.json` to store your app's complete description. Then, set the value of `description.full` in `manifest.json` as `$[file('description.txt')]`. Teams Toolkit reads the content from the text file and uses it as full description when building an app package.
```

File: msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md [hunk 11]
```diff
@@ -59,24 +59,68 @@ For example, you can put your manifest.json file in `test/test.json`, and update
...
+        "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.17/MicrosoftTeams.schema.json",
+        "manifestVersion": "1.17",
+        "description": {
+            "short": "Short description of tab",
+            "full": "$[file('./description.txt')]"
+        },
+    }
+   ```
+
+    You can also add the file path in `env/.env.xx`. Then, modify the parameter of file function to a placeholder in the `${{xx_xx}}` format.
```
Comment (v-npaladugu) [inline msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md:101]:
```suggestion
    You can also add the file path in `env/.env.xx`. Then, modify the parameter of `file()` to a placeholder in the `${{xx_xx}}` format.
```

**Review by v-npaladugu — CHANGES_REQUESTED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md](https://github.com/MicrosoftDocs/msteams-docs/blob/058369ea80733acc8dc22b88d234ea38ebd77e9c/msteams-platform%2Ftoolkit%2FTeamsFx-preview-and-customize-app-manifest.md) | 95 | 9 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/7e0087c8-2d2c-423d-bf04-0b4d1497b062) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Review by MuyangAmigo — APPROVED** [review]

**Review by Wajeed-msft — APPROVED** [review]

**Review by MSFT-yiz — APPROVED** [review]

**Review by MuyangAmigo — APPROVED** [review]

**Review by v-npaladugu — APPROVED** [review]

**Review by surbhigupta12 — APPROVED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/toolkit/TeamsFx-preview-and-customize-app-manifest.md](https://github.com/MicrosoftDocs/msteams-docs/blob/59c964e11a0833b693b6d4d478775d0eb401f631/msteams-platform%2Ftoolkit%2FTeamsFx-preview-and-customize-app-manifest.md) | 95 | 9 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/f59f6288-edab-49a1-ade4-b08c3792f025) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Pull request merged**